### PR TITLE
build(docs): Extract `build:redirects` into its own npm step, rather than running with `build:api-documentation`

### PR DESCRIPTION
**Pull request by Josmithr**
Separates concerns and resolves a potential race condition between the redirects build and the API docs build.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -12,17 +12,18 @@
 	"license": "MIT",
 	"author": "Microsoft and contributors",
 	"scripts": {
-		"build": "concurrently npm:build:api-documentation npm:build:md-magic && npm run hugo",
+		"build": "concurrently npm:build:api-documentation npm:build:md-magic npm:build:redirects && npm run hugo",
```
Comment (Josmithr) [inline docs/package.json:15]: There are still some unneeded overlaps between a number of the scripts here. I'm going to clean them up, though I'll do it in a separate PR.

**Review by RishhiB — APPROVED** [review]

**Comment by RishhiB** [comment]
Should we put these scripts in a folder within `docs/`?

**Comment by Josmithr** [comment]
> Should we put these scripts in a folder within `docs/`?

Like create a `docs/scripts` directory or something? I like that idea. Separate PR okay? (I'll put the download script in there too.

**Comment by Josmithr** [comment]
> > Should we put these scripts in a folder within `docs/`?
> 
> Like create a `docs/scripts` directory or something? I like that idea. Separate PR okay? (I'll put the download script, etc. in there too).

Actually, given how much work is being done in this area right now, I think I'll hold off on moving anything. But we should come back and do this once things have settled for sure.

**Pull request merged**